tpu_inference/layers/vllm/quantization/compressed_tensors/compressed_tensors_moe.py

```diff
@@ -0,0 +1,231 @@
+from typing import Callable, Optional, Union
+
+import jax
+import jax.numpy as jnp
+import torch
+import torch.nn.functional as F
+from compressed_tensors.quantization import QuantizationStrategy
+from jax.experimental.layout import Format, Layout
+from jax.sharding import Mesh, NamedSharding
+from jax.sharding import PartitionSpec as P
+from torch.nn.parameter import Parameter
+from torchax.interop import call_jax, torch_view
+from torchax.ops.mappings import t2j
+from vllm.logger import init_logger
+from vllm.model_executor.layers.fused_moe import FusedMoE, FusedMoEConfig
+from vllm.model_executor.layers.quantization.compressed_tensors.compressed_tensors import \
+    CompressedTensorsConfig
+from vllm.model_executor.layers.quantization.compressed_tensors.compressed_tensors_moe import \
+    CompressedTensorsW8A8Fp8MoEMethod
+from vllm.model_executor.layers.quantization.compressed_tensors.schemes.compressed_tensors_wNa16 import (  # noqa
+    WNA16_SUPPORTED_BITS, WNA16_SUPPORTED_TYPES_MAP)
+
+from tpu_inference.layers.vllm.quantization.common import JaxCommonConfig
+
+logger = init_logger(__name__)
+
+
+class VllmCompressedTensorsW8A8Fp8MoEMethod(CompressedTensorsW8A8Fp8MoEMethod,
+                                            JaxCommonConfig):
+
+    def __init__(self, quant_config: "CompressedTensorsConfig",
+                 moe: FusedMoEConfig, mesh: Mesh):
+        super().__init__(quant_config, moe)
+
+        self.use_marlin = False
+        self.use_cutlass = False
+        self.is_fp8_w8a8_sm100 = False
+
+        self.mesh = mesh
+        self.quant_config = quant_config
+        self.weight_quant = self.quant_config.target_scheme_map["Linear"].get(
```

Comment:
I don't think many of codes aren't needed anymore since many of these are taken care of during `CompressedTensorsW8A8Fp8MoEMethod` initialization which happens when `super().__init__(quant_config, moe)` is called.